File: Behavorial/Template/Template_py/Sandwich.py
import abc

class Sandwich(metaclass=abc.ABCMeta):
    def makeSandwich(self):
        self.addBread()
        self.addSauce();
        self.addStuffing()

    @abc.abstractmethod
    def addBread(self):
        pass

    @abc.abstractmethod
    def addSauce(self):
        pass

    @abc.abstractmethod
    def addStuffing(self):
        pass

File: Behavorial/Template/Template_Java/Sandwich.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Template_Java;

/**
 *
 * @author sashant
 */
public abstract class Sandwich {
   
    public void makeSandwich(){
        AddBread();
        AddSauce();
        AddStuffing();
        isGrilled();
    }
    
   public abstract void AddBread();
   public abstract void AddSauce();
   public abstract void AddStuffing();
   public abstract boolean isGrilled();
}
